daemon/volumes_unix.go

```diff
@@ -12,19 +12,37 @@ import (
 	mounttypes "github.com/docker/docker/api/types/mount"
 	"github.com/docker/docker/container"
 	volumemounts "github.com/docker/docker/volume/mounts"
+	"github.com/hashicorp/go-multierror"
 	"github.com/pkg/errors"
 )
 
 // setupMounts iterates through each of the mount points for a container and
 // calls Setup() on each. It also looks to see if is a network mount such as
 // /etc/resolv.conf, and if it is not, appends it to the array of mounts.
-func (daemon *Daemon) setupMounts(c *container.Container) ([]container.Mount, error) {
+func (daemon *Daemon) setupMounts(c *container.Container) ([]container.Mount, func() error, error) {
+	var cleanups []func() error
+	cleanup := func() error {
+		var allErr error
+		for _, c := range cleanups {
+			err := c()
+			if err != nil {
+				if allErr == nil {
+					allErr = err
+				} else {
+					allErr = multierror.Append(allErr, err)
+				}
+			}
+		}
+
+		return allErr
```

Comment:
Nit: I don't think it's worth the effort to avoid returning a multierror wrapping one error.
```suggestion
		var errs *multierror.Error
		for _, c := range cleanups {
			errs = multierror.Append(errs, c())
		}
		return errs.ErrorOrNil()
```